server/datastore/mysql/nanomdm_storage_test.go

```diff
@@ -83,3 +89,238 @@ func testEnqueueDeviceLockCommand(t *testing.T, ds *Datastore) {
+		ds:     ds,
+	}
+
+	// Number of concurrent lock attempts
+	numGoroutines := 20
+	var wg sync.WaitGroup
+	wg.Add(numGoroutines)
+
+	// Track successful locks
+	successCount := 0
+	var successMu sync.Mutex
+
+	// Track conflict errors
+	conflictCount := 0
+
+	// Collect all PINs that were generated
+	var pins []string
+
+	// Barrier to ensure all goroutines start at the same time
+	barrier := make(chan struct{})
+
+	for i := 0; i < numGoroutines; i++ {
+		go func(idx int) {
+			defer wg.Done()
+
+			// Wait for barrier
+			<-barrier
+
+			// Create a unique command for this goroutine
+			cmdUUID := fmt.Sprintf("test-lock-%03d", idx)
+			pin := fmt.Sprintf("%06d", 100000+idx) // Unique PIN for each request
+
+			cmd := &mdm.Command{
+				CommandUUID: cmdUUID,
+				Command: struct {
+					RequestType string
+				}{
+					RequestType: "DeviceLock",
+				},
+				Raw: []byte(fmt.Sprintf(`<?xml version="1.0"?><plist><dict><key>PIN</key><string>%s</string></dict></plist>`, pin)),
+			}
+
+			// Try to enqueue the lock command
+			err := storage.EnqueueDeviceLockCommand(ctx, host, cmd, pin)
+
+			switch {
+			case err == nil:
+				successMu.Lock()
+				successCount++
+				pins = append(pins, pin)
+				successMu.Unlock()
+			case fleet.IsConflict(err):
+				successMu.Lock()
+				conflictCount++
+				successMu.Unlock()
+			default:
+				// Unexpected error
+				t.Logf("Request %d got unexpected error: %v", idx, err)
+			}
+		}(i)
+	}
+
+	// Release all goroutines at once
+	close(barrier)
+
+	// Wait for all to complete
+	wg.Wait()
+
+	// Check the database state
+
+	// 1. Count how many DeviceLock commands were created
+	var commandCount int
+	err = ds.writer(ctx).Get(&commandCount,
+		`SELECT COUNT(*) FROM nano_commands WHERE command_uuid LIKE 'test-lock-%'`)
+	require.NoError(t, err)
+
+	// 2. Check what's stored in host_mdm_actions
+	var storedPIN string
+	var lockRef string
+	err = ds.writer(ctx).QueryRow(
+		`SELECT COALESCE(unlock_pin, ''), COALESCE(lock_ref, '') FROM host_mdm_actions WHERE host_id = ?`,
+		host.ID).Scan(&storedPIN, &lockRef)
+	require.NoError(t, err)
+
+	// Log the results
+	t.Logf("===== RACE CONDITION TEST RESULTS =====")
+	t.Logf("Concurrent requests sent: %d", numGoroutines)
+	t.Logf("Successful lock commands: %d", successCount)
+	t.Logf("Conflict errors: %d", conflictCount)
+	t.Logf("Commands in nano_commands table: %d", commandCount)
+	t.Logf("Final PIN stored in database: %s", storedPIN)
+	t.Logf("Final lock_ref in database: %s", lockRef)
+
+	// Assertions - only one lock should succeed
+	require.Equal(t, 1, successCount, "Only one lock command should succeed")
+	require.Equal(t, numGoroutines-1, conflictCount, "All other requests should get conflict error")
+	require.Equal(t, 1, commandCount, "Only one command should be in nano_commands table")
+	require.Len(t, pins, 1, "Only one PIN should be generated")
+	if len(pins) > 0 {
+		require.Equal(t, pins[0], storedPIN, "Stored PIN should match the successful request")
+	}
```

Comment:
nit: We can remove this length check since we do a length require check right above.